src/components/InventoryV2/InventoryListByNursery.tsx

```diff
@@ -0,0 +1,193 @@
+  },
+  spinnerContainer: {
+    position: 'fixed',
+    top: '50%',
+    left: '50%',
+  },
+}));
+
+const columns = (): TableColumnType[] => [
+  { key: 'facility_name', name: strings.NURSERY, type: 'string' },
+  {
+    key: 'speciesNames',
+    name: strings.SPECIES,
+    type: 'string',
+    tooltipTitle: strings.TOOLTIP_SCIENTIFIC_NAME,
+  },
+  {
+    key: 'germinatingQuantity',
+    name: strings.GERMINATING,
+    type: 'string',
+    tooltipTitle: strings.TOOLTIP_GERMINATING_QUANTITY,
+  },
+  {
+    key: 'notReadyQuantity',
+    name: strings.NOT_READY,
+    type: 'string',
+    tooltipTitle: strings.TOOLTIP_NOT_READY_QUANTITY,
+  },
+  { key: 'readyQuantity', name: strings.READY, type: 'string', tooltipTitle: strings.TOOLTIP_READY_QUANTITY },
+  { key: 'totalQuantity', name: strings.TOTAL, type: 'string', tooltipTitle: strings.TOOLTIP_TOTAL_QUANTITY },
+];
+
+export default function InventoryListByNursery() {
+  const theme = useTheme();
+  const classes = useStyles();
+  const { selectedOrganization } = useOrganization();
+  const [searchResults, setSearchResults] = useState<SearchResponseElement[] | null>();
+  const [unfilteredInventory, setUnfilteredInventory] = useState<SearchResponseElement[] | null>(null);
+  const [temporalSearchValue, setTemporalSearchValue] = useState('');
+  const debouncedSearchTerm = useDebounce(temporalSearchValue, 250);
+
+  const [nurseries, setNurseries] = useState<Facility[]>([]);
+  const [searchSortOrder, setSearchSortOrder] = useState<SearchSortOrder | undefined>({
+    field: 'species_scientificName',
+    direction: 'Ascending',
+  });
+  const [filters, setFilters] = useForm<InventoryFiltersType>({});
+
+  const { user } = useUser();
+  const numberFormatter = useNumberFormatter();
+  const numericFormatter = useMemo(() => numberFormatter(user?.locale), [user?.locale, numberFormatter]);
+
+  useEffect(() => {
+    setNurseries(getAllNurseries(selectedOrganization));
+  }, [selectedOrganization]);
+
+  const onSearchSortOrder = (order: SearchSortOrder) => {
+    const isClientSorted = BE_SORTED_FIELDS.indexOf(order.field) === -1;
+    setSearchSortOrder(isClientSorted ? undefined : order);
+  };
+
+  const onApplyFilters = useCallback(async () => {
+    const requestId = Math.random().toString();
+    setRequestId('searchInventory', requestId);
+    const apiSearchResults = await NurseryInventoryService.searchInventoryByNursery({
+      organizationId: selectedOrganization.id,
+      query: debouncedSearchTerm,
+      facilityIds: filters.facilityIds || nurseries.map((nr) => nr.id),
+      searchSortOrder,
+    });
+
+    let updatedResult: InventoryResultWithSpeciesNames[] | undefined = [];
+
+    const nextResults = apiSearchResults?.reduce((acc, result) => {
+      const resultTyped = result as FacilityInventoryResult;
+      const indexFound = acc.findIndex((savedResult) => savedResult.facility_id === resultTyped.facility_id);
+
+      if (indexFound !== undefined && indexFound !== -1) {
```

Comment:
This can also be `if (indexFound > -1) {`. Works as expected for `undefined` and `-1`